### PR TITLE
Fix #11540: Picklist trigger input changes on transfer

### LINKED ISSUE #11540: PickList: fire 'change' event
### Description

`<p:pickList>` doesn't fire any change event when an item is transferred.

### Describe the solution you would like

One way would be to add to `fireTransferEvent` function in `picklist.js`:
```
this.targetInput.trigger("change");
// and probably also this.sourceInput.trigger("change")?
```

### Additional context

This way we can safely reuse existing global form/input change event listeners on them.

### PR DESCRIPTION
**Pull request by melloware**
Fix #11540: Picklist trigger input changes on transfer

### CODE REVIEW

**Review by FlipWarthog — APPROVED** [review]

File: primefaces/src/main/resources/META-INF/resources/primefaces/picklist/picklist.js [hunk 1]
```diff
@@ -1168,6 +1168,10 @@ PrimeFaces.widget.PickList = PrimeFaces.widget.BaseWidget.extend({
             $($this.jqId + ' ul').sortable('enable');
             $this.updateButtonsState();
         }
+
+        // #11540: trigger change events on inputs for form handling
+        this.targetInput.trigger("change");
+        this.sourceInput.trigger("change")?
```
Comment (BalusC) [inline primefaces/src/main/resources/META-INF/resources/primefaces/picklist/picklist.js]: Question mark should be a semicolon.

Comment (jepsar) [inline primefaces/src/main/resources/META-INF/resources/primefaces/picklist/picklist.js]:
```suggestion
        this.sourceInput.trigger("change");
```

**Comment by melloware** [comment]
Thanks!

**Comment by jepsar** [comment]
Shouldn't this be added to the docs?

**Review by BalusC — APPROVED** [review]

**Comment by tandraschko** [comment]
if "transfer" triggers onchange, "reorder" should to
so it means if a user needs different listeners he uses transfer+reordern, change catches both

**Comment by BalusC** [comment]
You could trigger a namespaced event.

**Comment by jepsar** [comment]
I was thinking of https://github.com/primefaces/primefaces/blob/master/docs/14_0_0/components/picklist.md#ajax-behavior-events not sure if we should mention this there.

**Comment by melloware** [comment]
Hmm this just triggers the client side change events on the input but doesn't send an AJAX `change` event.  However I did realize that no `defaultEvent` was defined on this component and there should be so I made it `transfer`

**Pull request merged**